SPECS/annobin/annobin.spec

```diff
@@ -34,29 +21,189 @@ URL:     https://fedoraproject.org/wiki/Toolchain/Watermark
+%bcond_without clangplugin
+
+# Use "--without gccplugin" to disable the building of the annobin plugin for GCC.
+%bcond_without gccplugin
+
+# Use "--without llvmplugin" to disable the building of the annobin plugin for LLVM.
+%bcond_without llvmplugin
+
 # Set this to zero to disable the requirement for a specific version of gcc.
 # This should only be needed if there is some kind of problem with the version
 # checking logic or when building on RHEL-7 or earlier.
-%global with_hard_gcc_version_requirement 1
-
-# Enable this if it is necessary to build annobin without using annobin.
-# This is useful for example if the annobin plugin fails because of a change
-# in the size of gcc's global_options structure.  In order to rebuild annobin
-# against the changed gcc it is necessary to disable annobin as otherwise
-# the configuration step of annobin's build will fail.
-%undefine _annotated_build
+#
+# Update: now that we have gcc version checking support in redhat-rpm-config
+# there is no longer a great need for a hard gcc version check here.  Not
+# enabling this check greatly simplifies the process of installing a new major
+# version of gcc into the buildroot.
+%global with_hard_gcc_version_requirement 0
+
+%bcond_without plugin_rebuild
+# Allow the building of annobin without using annobin itself.
+# This is because if we are bootstrapping a new build environment we can have
+# a new version of gcc installed, but without a new of annobin installed.
+# (i.e. we are building the new version of annobin to go with the new version
+# of gcc).  If the *old* annobin plugin is used whilst building this new
+# version, the old plugin will complain that version of gcc for which it
+# was built is different from the version of gcc that is now being used, and
+# then it will abort.
+#
+# The default is to use plugin during rebuilds (cf BZ 1630550) but this can
+# be changed because of the need to be able to rebuild annobin when a change
+# to gcc breaks the version installed into the buildroot.  Note however that
+# uncommenting the lines below will result in annocheck not passing the rpminspect
+# tests....
+# %%if %%{without plugin_rebuild}
+# %%undefine _annotated_build
+# %%endif
 
 #---------------------------------------------------------------------------------
-Source:  https://nickc.fedorapeople.org/annobin-%{version}.tar.xz
+
+%global annobin_sources annobin-%{version}.tar.xz
+Source: https://nickc.fedorapeople.org/%{annobin_sources}
 # For the latest sources use:  git clone git://sourceware.org/git/annobin.git
 
-# Insert patches here, if needed.
-# Patch01: annobin-xxx.patch
+# This is where a copy of the sources will be installed.
+%global annobin_source_dir %{_usrsrc}/annobin
+
+# Insert patches here, if needed.  Eg:
+# Patch01: annobin-plugin-default-string-notes.patch
 
 #---------------------------------------------------------------------------------
 
+# Make sure that the necessary sub-packages are built.
+
+%if %{with gccplugin}
```

Comment:
these if statements will always be true as I said above. so probably can just be removed.